src/einsteinpy/symbolic/ricci.py

```diff
@@ -96,3 +152,44 @@ def symbols(self):
 
         """
         return self.syms
+
+
+class RicciScalar:
+    """
+        Class for defining Ricci Scalar
+        """
+
+    def __init__(self, metric_tensor, ricci_tensor):
+        if isinstance(metric_tensor, MetricTensor) and isinstance(
+            ricci_tensor, RicciTensor
+        ):
+            self.ricci_scalar = self.calculate_ricci_scalar(metric_tensor, ricci_tensor)
+
+        else:
+            raise TypeError(
+                "metric_tensor should be a MetricTensor class and ricci_tensor should be a RicciTensor"
+            )
+
+    def calculate_ricci_scalar(self, metric_tensor, ricci_tensor):
+        if not metric_tensor.config == "uu":
+            metric_tensor.change_config()
+        if not ricci_tensor.config == "ll":
+            ricci_tensor.change_config("ll")
+        identity = sympy.eye(ricci_tensor.dims)
+        ricci_temp = ricci_tensor.arr.tomatrix()
+        metric_temp = metric_tensor.arr.tomatrix()
+        aux = (identity * ricci_temp) * (identity * metric_temp)
+        ricci_scalar = aux.det()
```

Comment:
There seems to be a misunderstanding here as this is not the correct method for calculating the Ricci scalar. The first indication of such a misunderstanding is the usage of ``identity`` as it has no effect on the calculation of ``aux``. Secondly, the determinant of a matrix product is not the same as contracting indices of a tensor.

An appropriate calculation would instead utilize the functions ``tensorproduct`` and ``tensorcontraction`` provided by sympy:
```python
aux = tensorcontraction(tensorproduct(metric_temp, ricci_temp), (1,2))
aux = tensorcontraction(tensorproduct(metric_temp, aux), (1, 3))
ricci_scalar = tensorcontraction(aux, (0,1))
```

Reply:
What happens is that I look for a sympy function that would give me the elements of the diagonal, when I did not find it I looked for another alternative, even knowing that this was not the most appropriate calculation. So I'm going to take a look of what you've suggested! Thanks!